cuegui/cuegui/MenuActions.py

```diff
@@ -907,6 +907,34 @@ def stagger(self, rpcObjects=None):
         self.cuebotCall(__layer.staggerFrames, "Stagger Frames Failed",
                         frameRange, int(increment))
 
+    previewMain_info = ["Preview Main", None, "image"]
+
+    # pylint: disable=broad-except
+    def previewMain(self, rpcObjects=None):
+        try:
+            layer = self._getOnlyLayerObjects(rpcObjects)[0]
+            if layer is not None:
+                outputs = layer.getOutputPaths()
+                if len(outputs) > 0:
+                    cuegui.Utils.previewOutputs([outputs[0]])
```

Comment:
Can `self._getOnlyLayerObjects` return an empty list? If this is not a valid precondition, please add a comment stating that accessing the first position of the returned list is always safe.

Besides that, consider using `if outputs` instead of `if len(outputs) > 0`. I'm personally not a huge fan of this "idiom" but it's [python's standard way of checking for emptiness](https://docs.python.org/3/library/stdtypes.html#truth-value-testing).

```suggestion
            layer_objects = self._getOnlyLayerObjects(rpcObjects)
            if selected_layers:
                # Only displays the first item selected on the layer widget
                layer = selected_layers[0]
                outputs = layer.getOutputPaths()
                if outputs:
                    cuegui.Utils.previewOutputs([outputs[0]])
```

Reply:
I think if comes from working with the SG API, that has it's way of returning `None` results in strange ways :)